### ISSUE TITLE
"Dial tcp: i/o timeout" error when trying to read file

### ISSUE THREAD

**Issue by arthurpaimarnold** [colinmarc/hdfs#97]
The `cmd/hdfs/hdfs` executable is capable of connecting to one of our HDFS instances and listing directories normally. However, when we try to read a file (`./hdfs cat ...`), it prints

    dial tcp <IP>:50010: i/o timeout

This has been observed with other HDFS instances too.

**Comment by arthurpaimarnold** [comment]
Hi @colinmarc, 

I found the issue to be related to the `connectionTimeout` constant, which is defined to 1 second. If I increase it to 1 minute, I get a "no route to host" error, which is what I actually expected in this case.

Could you maybe make parameterize this value so users can set it themselves? Otherwise, could you maybe just increase it to 10 or 20 seconds?

Thanks

**Comment by colinmarc** [comment]
Yeah, they should be configurable. There's already an issue for that: https://github.com/colinmarc/hdfs/issues/76
